torchbenchmark/score/compute_score.py

```diff
@@ -7,32 +7,137 @@
 import math
 import sys
 import os
+import re
 import yaml
+import importlib
 
 from tabulate import tabulate
+from pathlib import Path
+from collections import defaultdict
 
-SPEC_FILE_DEFAULT = "score.yml"
+SPEC_FILE_DEFAULT = "torchbenchmark/score/score.yml"
 TARGET_SCORE_DEFAULT = 1000
+TORCHBENCH_V0_REF_DATA = "torchbenchmark/score/torchbench_0.0.yaml"
+TORCHBENCH_V1_REF_DATA = "" # Placeholder for v1 score
 
-def compute_score(config, data, fake_data=None):
-    target = config['target']
-    score = 0.0
-    weight_sum = 0.0
-    for name in config['benchmarks']:
-        cfg = config['benchmarks'][name]
-        weight, norm = cfg['weight'], cfg['norm']
-        weight_sum += weight
-        measured_mean = [b['stats']['mean'] for b in data['benchmarks'] if b['name'] == name]
-        assert len(measured_mean) == 1, f"Missing data for {name}, unable to compute score"
-        measured_mean = measured_mean[0]
-        if fake_data is not None and name in fake_data:
-            # used for sanity checks on the sensitivity of the score metric
-            measured_mean = fake_data[name]
-        benchmark_score = weight * math.log(norm / measured_mean)
-        # print(f"{name}: {benchmark_score}")
-        score += benchmark_score
-
-    score = target * math.exp(score)
-    assert abs(weight_sum - 1.0) < 1e-6, f"Bad configuration, weights don't sum to 1, but {weight_sum}"
-    return score
+def _get_model_task(model_name):
+    """
+    Helper function which extracts the task the model belongs to
+    by iterating over the Model attributes.
+    """
+    MODEL_PATH="../../models"
+    p = Path(__file__).parent.joinpath(MODEL_PATH + model_name)
```

Comment:
oh. i guess parent does exist!  

wait, i'm confused - isn't a better way here
p = Path(__file__).parent.parent / "models" / model_name
?
I don't like the "../../" explicit string if avoidable, and i'm not sure why you go 2 levels higher than parent?  shouldn't it be one level higher than parent?

Reply:
oh, backing up a step, it doesn't seem like you need to even check the path exists.

you aren't really being safe by checking the path exists and then doing import_module.  import_module could still fail, if for some reason the model code is broken.  So I'd suggest just skip the path exists check, and wrap import_module in a try: /except, raising a RuntimeError(f"Unable to get task for model {model_name}")

Reply:
Agreed and fixed.